internal/pkg/gateway/endorsement.go

```diff
@@ -102,25 +109,41 @@ func (p *plan) update(endorser *endorser, endorsement *peer.ProposalResponse) []
 		}
 	}
 
+	// check the proposal responses are the same
```

Comment:
Unrelated to this PR, but - in line 100 you get back from a peer an endorsement but its PKI-ID that you reference, was not computed from that peer's certificate, but it was computed based on what discovery told you. 

However, you may actually get a different certificate from what discovery told you, due to either the peer being malicious or the peer being behind a load balancer and then this `group` will be the empty string, and endorsers a nil slice. 

I suggest that instead of looking at the endorser which you computed with discovery, you will lookup using the endorsement which represents the reality and not the intent.

Reply:
Ok, that makes sense.  It's only being used here to cross reference which endorsing peers have responded so they can be removed from the layout(s).  However, given what you've said, it looks like I need to change the registry to use an ID that's not PKI-ID.  I'll raise a separate issue for this.